### PR TITLE
Boundary error image fix

### PR DESCRIPTION
**Pull request by banklesss**

### CODE REVIEW

File: src/components/Boundary/Boundary.tsx [hunk 1]
```diff
@@ -54,39 +56,58 @@ export const LoadingFallback = ({size = 'large', color = 'black', style}: Loadin
 )
 
 type ErrorBoundaryProps = {
-  error?: {fallback?: ReactErrorBoundaryProps['fallbackRender']; enabled?: boolean}
+  error?: {
+    fallback?: ReactErrorBoundaryProps['fallbackRender']
+    enabled?: boolean
+    notFill?: ErrorFallbackProps['notFill']
```
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx:62]: no

File: src/components/Boundary/Boundary.tsx [hunk 2]
```diff
@@ -54,39 +56,58 @@ export const LoadingFallback = ({size = 'large', color = 'black', style}: Loadin
...
-export const ErrorFallback = ({error, resetErrorBoundary, reset = true}: ErrorFallbackProps) => (
-  <View style={styles.container}>
-    <View style={styles.errorHeader}>
-      <Text>{error.message}</Text>
-    </View>
+type ErrorFallbackProps = {
+  error: FallbackProps['error'] & MessageDescriptor
+  resetErrorBoundary: FallbackProps['resetErrorBoundary']
+  reset?: boolean
+  notFill?: boolean
```
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx:83]: no

File: src/components/Boundary/Boundary.tsx [hunk 3]
```diff
@@ -54,39 +56,58 @@ export const LoadingFallback = ({size = 'large', color = 'black', style}: Loadin
...
 const ErrorBoundary = ({children, ...props}: ErrorBoundaryProps) => {
   if (props.error?.enabled === false) return <>{children}</>
 
   const fallbackRender = (fallbackProps: ErrorFallbackProps) =>
-    props.error?.fallback?.(fallbackProps) || <ErrorFallback {...fallbackProps} />
+    props.error?.fallback?.(fallbackProps) || <ErrorFallback {...fallbackProps} notFill={props.error?.notFill} />
 
-  return <ReactErrorBoundary fallbackRender={fallbackRender}>{children}</ReactErrorBoundary>
+  return (
+    <LanguageProvider>
```
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx:73]: no

File: src/components/Boundary/Boundary.tsx (hunk 2 above)
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx]: avoid props which directly relate to rendering ui

Comment (wolverineks) [reply src/components/Boundary/Boundary.tsx]: no margin, padding, etc...

File: src/components/Boundary/Boundary.tsx (hunk 3 above)
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx]: what is this for?

**Review by wolverineks — CHANGES_REQUESTED** [review]
comments

File: src/components/Boundary/Boundary.tsx (hunk 3 above)
Comment (banklesss) [reply src/components/Boundary/Boundary.tsx]: to be able to get the error messages: https://github.com/Emurgo/yoroi-mobile/blob/b7f5f987e29c1e854f31b942daa3bf2ba7fca910/src/legacy/errors.ts#L22

File: src/components/Boundary/Boundary.tsx [hunk 4]
```diff
@@ -80,13 +80,13 @@ type ErrorFallbackProps = {
...
-  notFill?: boolean
+  size?: 'small' | 'large' | undefined
 }
 
-export const ErrorFallback = ({error, resetErrorBoundary, reset = true, notFill = false}: ErrorFallbackProps) => {
+export const ErrorFallback = ({error, resetErrorBoundary, reset = true, size = 'large'}: ErrorFallbackProps) => {
   const intl = useIntl()
   return (
-    <View style={[styles.container, {flex: notFill === false ? 1 : undefined}]}>
+    <View style={[styles.container, {flex: size === 'large' ? 1 : undefined}]}>
```
Comment (stackchain) [inline src/components/Boundary/Boundary.tsx]:
```suggestion
    <View style={[styles.container, size === 'large' && {flex: 1}]}>
```

**Review by stackchain — CHANGES_REQUESTED** [review]

**Review by stackchain — APPROVED** [review]

File: src/TxHistory/TxHistory.stories.tsx [hunk 5]
```diff
@@ -13,6 +14,21 @@ storiesOf('V2/TxHistory', module)
       </SelectedWalletProvider>
     )
   })
+  .add('default with API errors', () => {
+    const wallet = {
+      ...mockWallet,
+      fetchCurrentPrice: () => Promise.reject(new Error('fetchCurrentPrice failed')),
+      fetchUTXOs: () => Promise.reject(new Error('fetchUTXOs failed')),
```
Comment (stackchain) [inline src/TxHistory/TxHistory.stories.tsx:24]: Since the Boundary is using `intl` you will need to add the i18n decorator as well.

**Review by stackchain — COMMENTED** [review]
comment

File: src/TxHistory/TxHistory.stories.tsx (hunk 5 above)
Comment (stackchain) [reply src/TxHistory/TxHistory.stories.tsx:24]: also to the `Boundary.stories`

Comment (wolverineks) [reply src/TxHistory/TxHistory.stories.tsx:24]: LanguageProvider wraps the entire app

Comment (banklesss) [reply src/TxHistory/TxHistory.stories.tsx:24]: Is the question related to consider the cases where the errors messages are generated with `react-intl` @stackchain ?

File: src/components/Boundary/Boundary.tsx [hunk 6]
```diff
@@ -54,39 +56,106 @@ export const LoadingFallback = ({size = 'large', color = 'black', style}: Loadin
...
   return <ReactErrorBoundary fallbackRender={fallbackRender}>{children}</ReactErrorBoundary>
 }
 
-type ErrorFallbackProps = FallbackProps & {reset?: boolean}
-export const ErrorFallback = ({error, resetErrorBoundary, reset = true}: ErrorFallbackProps) => (
-  <View style={styles.container}>
-    <View style={styles.errorHeader}>
-      <Text>{error.message}</Text>
+type ErrorFallbackProps = {
+  error: FallbackProps['error'] & MessageDescriptor
```
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx]: `error: FallbackProps['error'] | LocalizableError`

File: src/components/Boundary/Boundary.tsx [hunk 7]
```diff
@@ -54,39 +56,106 @@ export const LoadingFallback = ({size = 'large', color = 'black', style}: Loadin
...
+        </Text>
+      </TouchableOpacity>
+    </View>
+  )
+}
 
 const styles = StyleSheet.create({
-  container: {flex: 1, alignItems: 'center', justifyContent: 'center'},
-  errorHeader: {alignItems: 'center', justifyContent: 'center', padding: 16},
+  largeContainer: {alignItems: 'center', justifyContent: 'center', flex: 1},
```
Comment (wolverineks) [inline src/components/Boundary/Boundary.tsx]: `flex: 1` breaks WithdrawStakingRewards error

**Review by wolverineks — CHANGES_REQUESTED** [review]
comments

File: src/components/Boundary/Boundary.stories.tsx [hunk 8]
```diff
@@ -2,23 +2,26 @@ import {storiesOf} from '@storybook/react-native'
 import React from 'react'
 import {Button, StyleSheet, Text, View} from 'react-native'
 import {ActivityIndicator} from 'react-native-paper'
-import {QueryClient, QueryClientProvider, useQuery} from 'react-query'
+import {useQuery} from 'react-query'
 
+import {QueryProvider} from '../../../storybook/decorators/query'
```
Comment (stackchain) [inline src/components/Boundary/Boundary.stories.tsx]: export from `index.ts`
```suggestion
import {QueryProvider} from '../../../storybook/decorators'
```

**Review by stackchain — CHANGES_REQUESTED** [review]

File: src/components/Boundary/Boundary.stories.tsx (hunk 8 above)
Comment (banklesss) [reply src/components/Boundary/Boundary.stories.tsx]: done thanks

**Review by wolverineks — APPROVED** [review]

**Pull request merged**